x/slashing/keeper.go

```diff
@@ -55,8 +55,12 @@ func (k Keeper) handleValidatorSignature(ctx sdk.Context, pubkey crypto.PubKey,
 	address := pubkey.Address()
 
 	// Local index, so counts blocks validator *should* have signed
-	// Will use the 0-value default signing info if not present
-	signInfo, _ := k.getValidatorSigningInfo(ctx, address)
+	// Will use the 0-value default signing info if not present, except for start height
+	signInfo, found := k.getValidatorSigningInfo(ctx, address)
+	if !found {
+		// If this validator has never been seen before, set the start height
+		signInfo.StartHeight = height
+	}
```

Comment:
I tend to think this is bad practice, I think it would be more clear to have a `NewSignInfo` function where you pass in the height and we explicitly define signInfo with this function if `!found` - this way we can remove that second comment there that rationalizes what is happening - it will just be implied by the code

Reply:
Agreed - that's clearer - changed to `NewValidatorSigningInfo`.